### ISSUE TITLE
User watchlist

### ISSUE THREAD

**Issue by modem7** [sct/overseerr#327]
The ability to have a section where users are able to create a watchlist of things they want to watch.

It would be able to leverage features such as web link to plex (so people can click and it goes directly to the media in question), and request management (users can put items on the watchlist, but not necessarily request it yet).

1. User logs in and selects things the things they want to be on their list, both movies and TV.
2. Leverage Plex API to see what has already been watched in Plex (if this is not possible, users should be able to manually able to select watched/unwatched statuses).
3. When a user is ready to move onto the next series/movie, they are able to click "request" and it follows standard request processes.
4. Users will be able to fully manage their wish/watchlists with watched/unwatched/delete/add permissions.

There is currently no other system that is a watchlist *and* can leverage Plex/Radarr/Sonarr. The closest is Flox for the watchlist, but work has been stopped on that, and Ombi only has request management features.

**Comment by modem7** [comment]
Commenting so this doesn't get closed automatically

**Comment by WoisWoi** [comment]
In this perspective, it might be nice to be able to create recommended media lists for all users

**Comment by modem7** [comment]
bump to avoid stalebot

**Comment by modem7** [comment]
bump

**Comment by modem7** [comment]
bump.

Could we set a no stale label on this?

**Comment by modem7** [comment]
> In this perspective, it might be nice to be able to create recommended media lists for all users

Absolutely, it should be relatively simple to do that if this gets implemented.

**Comment by TheCatLady** [comment]
> > In this perspective, it might be nice to be able to create recommended media lists for all users
> 
> Absolutely agreed. It would give value add and create a portal not just for requests, but also for future requests which would increase usage of the platform.

See #2215; it's not something that is possible at this time.

**Comment by modem7** [comment]
> > > In this perspective, it might be nice to be able to create recommended media lists for all users
> > 
> > Absolutely agreed. It would give value add and create a portal not just for requests, but also for future requests which would increase usage of the platform.
> 
> See #2215; it's not something that is possible at this time.

In that particular use case, the reply certainly makes sense!

However, my particular request is effectively:

User has a list where they can add things they wish to request in future, but not necessarily now (e.g. They've requested season 1 of blah blah, and want to queue a request for season 2 and 3, or they want to save a list of movies that they wish to request, but don't want to put the request in now).

This would take the onus away from the admin having a substantial number of queued requests that aren't currently needed and/or takes the onus away from the user having to search xyz at a future point.

TLDR: A list per user where they can add things that they've searched, but don't yet wish to press "request". 

I hope that makes sense/things clearer?

**Comment by modem7** [comment]
bump

**Comment by jangerhard** [comment]
> Following the recent Plex [watchlist](https://www.plex.tv/en-ca/blog/end-the-streaming-struggle-with-plex/) feature release, could Overseerr integrate with it? (i.e. forward any new addition to the watchlist to Radarr/Sonarr)

Seconding this. We could treat every entry in the watchlist as a new request in Overseerr, filtered to remove entries that are already requested. Is the watchlist available from the plex-api yet?

**Comment by anonyme123** [comment]
Any thoughts from the developers?

**Comment by modem7** [comment]
Stale bump again.

**Comment by modem7** [comment]
Ok.... Apparently stalebot doesn't react to updates as intended.

Feel free to reopen if this is of interest.

**Comment by danshilm** [comment]
It is, sct is working on the Plex watchlist feature already.

**Comment by anonyme123** [comment]
@danshilm thanks for the 1.30 release. The watchlist sync job only runs every 10 min at the most frequent. Is there a way you could immediately forward any additions to the Plex watchlist to Sonarr/Radarr?

**Comment by Keelan** [comment]
There seems to be a check box in the permissions for viewing watch-lists, but i can't make out what that does in the UX. By asking someone to add something to their watchlist then searching overseerr i was able to confirm it works great but i have no history of who/when. Looking at a user who has made only plex watchlist request yeilds an empty request history.

Is there any plans to be able to see this in the users request history?

**Comment by danshilm** [comment]
> There seems to be a check box in the permissions for viewing watch-lists, but i can't make out what that does in the UX. By asking someone to add something to their watchlist then searching overseerr i was able to confirm it works great but i have no history of who/when. Looking at a user who has made only plex watchlist request yeilds an empty request history.
> 
> Is there any plans to be able to see these requests when looking at a users profile, just like requests made within overseerr?
> 
> 1.31.0 Stable

There's another setting on a user profile's general settings, "Auto-Request Movies/Series" that does just that. It runs on a schedule defined in the list of jobs on the Settings -> Jobs & Cache page, named "Plex Watchlist Sync". Users also need to be in your list of friends on Plex that have media shared with them and they also need to have logged in to Overseerr with their Plex account at least once.
